test/e2e/proxy_test.go

```diff
@@ -289,6 +289,50 @@ func TestProxyFlow(t *testing.T) {
 				_, err := hostAwaitWithShorterTimeout.CreateAPIProxyClient(t, user.token, proxyWorkspaceURL)
 				require.EqualError(t, err, `an error on the server ("unable to get target cluster: the requested space is not available") has prevented the request from succeeding`)
 			})
+
+			t.Run("invalid request headers", func(t *testing.T) {
+				// given
+				proxyWorkspaceURL := hostAwait.ProxyURLWithWorkspaceContext(user.compliantUsername)
+				rejectedHeaders := []headerKeyValue{
```

Comment:
Does it make sense to check also the other headers:

- Impersonate-User
- Impersonate-Extra-( extra name )
- Impersonate-Uid

and mixing them , as we did with the unit tests and as documented here https://kubernetes.io/docs/reference/access-authn-authz/authentication/#user-impersonation ?

Reply:
I tested other headers without the changes in https://github.com/codeready-toolchain/registration-service/pull/317 but couldn't get it to do what I was expecting so I figured it wasn't worth testing in e2e tests. They're covered in unit tests so that's good enough I think.